src/ceph-daemon/ceph-daemon

```diff
@@ -1265,13 +1377,18 @@ def command_bootstrap():
         logger.info('Enabling the dashboard module...')
         cli(['mgr', 'module', 'enable', 'dashboard'])
         logger.info('Waiting for the module to be available...')
-        # FIXME: potential for an endless loop?
+
+        limit = 20
         while True:
             c_out = cli(['-h'])
             if 'dashboard' in c_out:
                 break
-            logger.info('Dashboard not yet available, waiting...')
+            logger.info('Dashboard not yet available, waiting...({})'.format(limit))
             time.sleep(1)
+            limit -= 1
+            if limit == 0:
+                raise Error("Timeout waiting for dashboard to be available")
+
```

Comment:
put this in a separate patch?

Reply:
Point taken - but if we agree we need it, why not keep in and save a further PR and review process?

Reply:
@pcuzner please see PR #32049. Looks like we are attempting to solve the same issue, but I'm aok with either approach ...

Reply:
@mgfritch I'll back my change out, in favour of your approach.